homeassistant/components/group/entity.py

```diff
@@ -425,27 +465,47 @@ def _async_update_group_state(self, tr_state: State | None = None) -> None:
         elif tr_state.attributes.get(ATTR_ASSUMED_STATE):
             self._assumed_state = True
 
-        num_on_states = len(self._on_states)
+        # If we do not have an on state for any domains
+        # we use None (which will be STATE_UNKNOWN)
+        if (num_on_states := len(self._on_states)) == 0:
+            self._state = None
+            return
+
         # If all the entity domains we are tracking
         # have the same on state we use this state
         # and its hass.data[REG_KEY].on_off_mapping to off
+        group_is_on = self.mode(self._on_off.values())
         if num_on_states == 1:
```

Comment:
Please move calculating `group_is_on` above the comment
```suggestion
        group_is_on = self.mode(self._on_off.values())

        # If all the entity domains we are tracking
        # have the same on state we use this state
        # and its hass.data[REG_KEY].on_off_mapping to off
        if num_on_states == 1:
```